packages/react-devtools-shared/src/backend/profilingHooks.js

```diff
@@ -858,6 +878,26 @@ export function createProfilingHooks({
           snapshotHeight: 0,
         };
         nextRenderShouldStartNewBatch = true;
+      } else {
+        // Postprocess Profile data
+        if (currentTimelineData !== null) {
+          currentTimelineData.schedulingEvents.forEach(event => {
+            if (event.type === 'schedule-state-update') {
+              // TODO(luna): We can optimize this by creating a map of
+              // fiber to component stack instead of generating the stack
+              // for every fiber every time
+              const fiberStack = currentFiberStacks.get(event);
+              if (fiberStack) {
+                event.componentStack = fiberStack.reduce((trace, fiber) => {
```

Comment:
What if the tree is really deep? Maybe this would be better as an array instead of a string so we can show part of the tree instead of the whole thing 🤔

Reply:
How deep are you thinking? Is your concern more about serializing the large array/string or about the frontend not being able to display it?

If it's the former, we could add max depth logic here, like (`fiberStack.slice(0, SOME_MAX_DEPTH).reduce(...)`?) and then just prepend "..." to the string.

If it's the latter, seems like the frontend could use `max-height` and `overflow` styles to control how much stack it showed?

Reply:
Sounds good.